airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -0,0 +1,222 @@
+
+        params = {self.replication_start_date_field: self._replication_start_date, self.page_size_field: self.page_size}
+        if self._replication_start_date:
+            start_date = max(stream_state.get(self.cursor_field, self._replication_start_date), self._replication_start_date)
+            params.update({self.replication_start_date_field: start_date})
+        return params
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        stream_data = response.json()
+        next_page_token = stream_data.get("nextToken")
+        if next_page_token:
+            return {self.next_page_token_field: next_page_token}
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = response.json().get(self.data_field, [])
+        yield from records
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    def _get_access_token(self) -> str:
+        """
+        Get's the access token
+        :return: access_token str
+        """
+        data = {"grant_type": "refresh_token", **self._access_token_credentials}
+        headers = {"User-Agent": "python-sp-api-0.6.2", "content-type": "application/x-www-form-urlencoded;charset=UTF-8"}
+        res = requests.post("https://api.amazon.com/auth/o2/token", data, headers)
+        return res.json()["access_token"]
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {
+            "host": "sellingpartnerapi-na.amazon.com",
+            "user-agent": "python-sp-api-0.6.2",
+            "x-amz-access-token": self._get_access_token(),
+            "x-amz-date": pendulum.now("utc").strftime("%Y%m%dT%H%M%SZ"),
+            "content-type": "application/json",
+        }
+
+    def _create_prepared_request(
+        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None, auth: AWSSigV4 = None
+    ) -> requests.PreparedRequest:
+        args = {"method": self.http_method, "url": self.url_base + path, "headers": headers, "params": params, "auth": auth}
+
+        if self.http_method.upper() == "POST":
+            args["json"] = json
```

Comment:
Please insert `TODO` above this line from base class so in the future we won't forget to add support for non-json body in this source too.
![image](https://user-images.githubusercontent.com/19872253/126801654-7790a6a2-e9cd-4335-9193-8ea0b58e77c5.png)